### ISSUE TITLE
Phi-3 vision on Android

### ISSUE THREAD

**Issue by sreenivasraghavan71** [microsoft/onnxruntime-genai#751]
HI, I am trying to add the LLM functionality to the android compatible devices. Can anyone tell me how to build for the android? Also, is any help on the front of Multimodal LLM deployment on mobile would be really good?

**Comment by skyline75489** [comment]
If you want to build the package yourself:

```bash
# Assume you have correctly set envvar ANDROID_HOME and ANDROID_NDK_HOME
python build.py --android --android_api 31 --build_java
```

You'll find the AAR at `build/Android/RelWithDebInfo/src/java/build/android/outputs/aar`

**Comment by skyline75489** [comment]
There's also (WIP) Android example here: https://github.com/microsoft/onnxruntime-inference-examples/pull/449, if you're interested.

**Comment by sreenivasraghavan71** [comment]
HI, I am specifically trying to deploy multimodal model on android device. I tried the phi3- text model and it is working fine. However, the real issue starts with phi3-vision model. Is there any Java APIs for multimodal which i can use directly and any process to build the onnxruntime for android (not through python route,as mentioned above). Also, I am interested to know if the onnxruntime-genai capability can be extended to other open source models from hugging face?

**Comment by natke** [comment]
Hi @sreenivasraghavan71,

We have added multi modal support to the Java API. Right now the Phi-3 vision model only runs on CPU thought and is quite slow due to the long prompt created by the image input.

We are working on running the vision model on NPU too.

**Comment by natke** [comment]
Closed as a duplicate of #608
